src/test.js

```diff
@@ -74,8 +74,10 @@ Test.prototype = {
 		config.current = this;
 
 		if ( this.module.testEnvironment ) {
+			delete this.module.testEnvironment.before;
 			delete this.module.testEnvironment.beforeEach;
 			delete this.module.testEnvironment.afterEach;
+			delete this.module.testEnvironment.after;
```

Comment:
> Anyone previously extending that module with these keywords would get their tests affected.

Exactly. Code like `QUnit.module( name, { after: new Date(2016, 0, 1) }, … )`, _which is currently supported_, will be broken. This is why I abhor the conflation of module options and test context in a single parameter, but there it is. And it's different from adding `QUnit.skip` because `QUnit` is our object, but that parameter is explicitly wide open for users, minus our exceptions.

Anyway, while maybe not a huge deal, this is technically breaking.